utils/shared.js

```diff
@@ -1180,50 +1180,59 @@ const handleCancerOccurrences = async (incomingCancerOccurrenceArray, requiredOc
     const existingCancerOccurrences = await getParticipantCancerOccurrences(participantToken);
-    
+
     // Make sure the occurrence is not a duplicate - check timestamp and yes values in primary cancer sites.
     const duplicateCheckResponse = checkForDuplicateCancerOccurrences(incomingCancerOccurrenceArray, existingCancerOccurrences);
     if (duplicateCheckResponse.error) {
         return { error: true, message: duplicateCheckResponse.message, data: [] };
     }
 
     const finalizedCancerOccurrenceArray = [];
-    // Get the highest occurrence number. Each new occurrence gets a sequential occurrenceNumber (specific to the participant).
-    let highestOccurrenceNumber = getHighestOccurrenceNumber(existingCancerOccurrences);
+    let mostRecentOccurrenceNum = existingCancerOccurrences.length;
     for (const newOccurrence of incomingCancerOccurrenceArray) {
         // Finalize the occurrence data and add to the finalized array.
-        const occurrenceData = finalizeCancerOccurrenceData(newOccurrence, participantToken, participantConnectId, highestOccurrenceNumber);
+        const occurrenceData = finalizeCancerOccurrenceData(newOccurrence, participantToken, participantConnectId, mostRecentOccurrenceNum);
         finalizedCancerOccurrenceArray.push(occurrenceData);
-        highestOccurrenceNumber++;
+        mostRecentOccurrenceNum++;
     }
 
     return { error: false, message: 'Success!', data: finalizedCancerOccurrenceArray };
 }
 
 /**
- * Validate that at least one cancer site in the Occurrence object (740819233) has a value of 'yes'.
+ * Validate additional cancer site requirements in the Occurrence object (740819233).
+ * Important: fieldMapping.primaryCancerSiteObj value has already been validated in the initial validation step.
+ * If the 'fieldMapping.cancerSites.other' cancer site is selected, the 'fieldMapping.anotherTypeOfCancerText' field is required.
+ * Else, the 'anotherTypeOfCancerText' field should not be present.
  * @param {object} cancerSitesObject - property (740819233) in the cancer occurrence object (637153953).
- * @returns {boolean} - true if at least one cancer site has a value of 'yes'. false otherwise.
+ * @returns {boolean} - Returns true the above requirements are met, false otherwise.
  */
 const validateCancerOccurrence = (cancerSitesObject) => {
-    for (const key in cancerSitesObject) {
-        if (key === fieldMapping.cancerSites.anotherTypeOfCancer.toString() && cancerSitesObject[key] && typeof cancerSitesObject[key] === 'string') {
-            return true;
-        } 
-        else if (cancerSitesObject[key] === fieldMapping.yes) {
-            return true;
-        }
+    if (!cancerSitesObject || Object.keys(cancerSitesObject).length === 0 || !cancerSitesObject[fieldMapping.primaryCancerSiteCategorical]) {
+        return { error: true, message: 'Primary cancer site categorical (740819233.149205077) is required.', data: [] };
```

Comment:
Will `isError` or `hasError` a better key inside this returned object?

Reply:
I like your naming convention suggestion but wanted to stick with the naming convention pattern established in `updateParticipantData`, where this ultimately returns (updateParticipantData -> handleCancerOccurrences -> validateCancerOccurrence. Thanks!